### ISSUE TITLE
Failed install leaves broken package

### ISSUE THREAD

**Issue by joshkel** [yarnpkg/yarn#1539]
**Do you want to request a _feature_ or report a _bug_?**

Bug

**What is the current behavior?**

If running a package's install script fails, Yarn leaves the broken package on disk and fails to detect that it still has problems on subsequent runs.

**If the current behavior is a bug, please provide the steps to reproduce.**
1. Pick a package that has a non-trivial install script and ensure that its install script will fail. (Installing `phantomjs` with no `TMPDIR` or `TEMP` in your environment should meet this criteria - see #1538.)
2. Run `yarn` to install it. Note the error.
3. Run `yarn` again. It appears to work.

``` bash
# Create a sample project with a problematic package.
echo '{"devDependencies":{"phantomjs":"2.1.7"}}' > package.json

# Run Yarn. It will fail.
TMPDIR= TEMP= yarn

# Run Yarn again. It will appear to work, but the package is still not properly installed.
TMPDIR= TEMP= yarn
```

**What is the expected behavior?**

Running `yarn` the second time should give the same error as running it the first time. (Maybe it should track which packages still need to have their install scripts run, or maybe it should delete the package directory and mark it as not installed on error.)

**Please mention your node.js, yarn and operating system version.**

Node 4.6.1, Yarn 0.16.1, Ubuntu 14.04

**Comment by starwed** [comment]
This has bitten me several times on TravisCI -- it will cache a broken build of phantomjs, since yarn reports it as working on a retry.   Example [here](https://travis-ci.org/craftyjs/Crafty/builds/192542266#L473), where I get the message "Cannot read property 'pid' of undefined" due to a failure to start phantomjs.

**Comment by dcecile** [comment]
It looks like this has been fixed in `phantomjs-prebuilt` v2.1.8 via Medium/phantomjs@5e6598f.

(If you're finding this bug because you're using `prerender-spa-plugin`, *that* package has an open issue to upgrade from `phantomjs-prebuilt` v2.1.7 where it's currently stuck due to a Windows problem -- see chrisvfritz/prerender-spa-plugin#36.)

**Comment by merceyz** [comment]
Closing as fixed in v2

https://yarnpkg.com/getting-started/migration
